### PR TITLE
Fixes #3080 - Do not create and use jars with "tests" classifier.

### LINKED ISSUE #3080: Do not create and use jars with "tests" classifier
Some Jetty Maven modules create a "tests" jar using the `test-jar` goal of the Maven Jar Plugin.
These "tests" jars contain utility classes that are later reused by other modules.

Unfortunately these jars are very cumbersome to use with JPMS, as they need special command line options passed to the Maven Compiler Plugin and the Maven Surefire Plugin.
Not only, the additional JPMS command line options often require to explicitly add additional Maven dependencies (especially transitive ones) that would not be necessary otherwise, and that if missing lead to build errors that are difficult to figure out.

We should refactor the test utility classes into a separate proper Maven module, and use normal Maven test dependencies which will play well with JPMS.

### PR DESCRIPTION
**Pull request by sbordet**
Removed the jetty-server and jetty-servlet test-jar artifacts.
The only one left is the jetty-maven-plugin test-jar.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

**Review by joakime — CHANGES_REQUESTED** [review]
`ServletTester` is used by many projects, and several open source projects.
We cannot get rid of it.

Move it to another location, don't delete it.

File: jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletTester.java [hunk 1]
```diff
@@ -39,6 +40,51 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * <p>ServletTester should not be used.</p>
```
Comment (gregw) [inline jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletTester.java]: Perhaps say:
```suggestion
 * <p>ServletTester is not best practise and may be deprecated in future releases.</p>
```

File: jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletTester.java [hunk 2]
```diff
@@ -39,6 +40,51 @@
...
+ * ServletContextHandler context = new ServletContextHandler(server, "/contextPath");
+ * // Configure the context here.
+ * server.start();
+ * </pre>
+ * <p>You can configure the context by adding Servlets and Filters, attributes,
+ * etc. even after it has been started.
+ * Use {@link HttpTester} and {@link LocalConnector} to make HTTP/1.1 requests,
+ * in this way:</p>
+ * <pre>
+ * // Generate the request.
```
Comment (gregw) [inline jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletTester.java:63]: Perhaps show two examples.  One using LocalConnector directly with strings and then show HttpTester usage.

Comment (gregw) [reply jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletTester.java:63]: actually never mind...

**Review by gregw — APPROVED** [review]
other than slightly better wording. LGTM

**Pull request merged**